src/services/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package services.implementation;
+
+import entities.Client;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import exceptions.services.NullEntityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.interfaces.IClientInterface;
+import util.ClientServiceUtility;
+import util.UserUtil;
+
+public class ClientServiceImpl implements IClientInterface {
+  private final List<Client> clients;
+  private Long currentId;
+
+  public ClientServiceImpl() {
+    this.clients = new ArrayList<>();
+    this.currentId = 0L;
+  }
+
+  @Override
+  public Optional<Client> insert(Client client) {
+    client.setId(this.currentId);
+    client.setCode(ClientServiceUtility.createCode(this.currentId,"C"));
+    this.currentId=this.currentId+1;
+    this.clients.add(client);
+    return Optional.of(client);
+  }
+
+  @Override
+  public Optional<Client> findById(Long id) {
+    Optional.ofNullable(id).orElseThrow(IdNullPointerException::new);
+    var client =
+        this.clients.stream()
+            .filter(c -> id.equals(c.getId()) && !(c.getStatus() == StatusEnum.DELETED))
+            .findAny()
+            .orElseThrow(() -> new EntityNotFoundException(Client.class, id));
+    return Optional.ofNullable(client);
+  }
+
+  @Override
+  public Optional<Client> delete(Long id) {
+    Optional.ofNullable(id).orElseThrow(IdNullPointerException::new);
+    var client = this.findById(id).get();
+    client.setStatus(StatusEnum.DELETED);
+    return this.update(client);
+  }
+
+  @Override
+  public Optional<Client> update(Client object) {
```

Comment:
Change updated date to current date time

Reply:
solved